frontend/src/employee-frontend/components/person-profile/PersonVoucherValueDecisions.tsx

```diff
@@ -0,0 +1,102 @@
+import { formatCents } from 'employee-frontend/utils/money'
+
+interface Props {
+  id: string
+  open: boolean
+}
+
+export default React.memo(function PersonVoucherValueDecisions({
+  id,
+  open
+}: Props) {
+  const { i18n } = useTranslation()
+
+  const [voucherValueDecisions, setVoucherValueDecisions] = useState<
+    Result<VoucherValueDecisionSummary[]>
+  >(Loading.of())
+
+  const loadDecisions = useCallback(() => {
+    setVoucherValueDecisions(Loading.of())
+    void getPersonVoucherValueDecisions(id).then((response) => {
+      setVoucherValueDecisions(response)
+    })
+  }, [id, setVoucherValueDecisions])
+
+  useEffect(loadDecisions, [loadDecisions])
+
+  return (
+    <CollapsibleSection
+      icon={faChild}
+      title={i18n.personProfile.voucherValueDecisions}
+      data-qa="person-voucher-value-decisions-collapsible"
+      startCollapsed={!open}
+    >
+      <Table data-qa="table-of-voucher-value-decisions">
+        <Thead>
+          <Tr>
+            <Th>{i18n.valueDecisions.table.validity}</Th>
+            <Th>{i18n.valueDecisions.table.number}</Th>
+            <Th>{i18n.valueDecisions.table.totalValue}</Th>
+            <Th>{i18n.valueDecisions.table.totalCoPayment}</Th>
+            <Th>{i18n.valueDecisions.table.createdAt}</Th>
+            <Th>{i18n.valueDecisions.table.sentAt}</Th>
+            <Th>{i18n.valueDecisions.table.status}</Th>
+          </Tr>
+        </Thead>
+        {voucherValueDecisions
+          .map((data) => (
+            <Tbody key="table-body">
+              {orderBy(data, ['createdAt'], ['desc']).map((decision) => {
+                return (
+                  <Tr
+                    key={`${decision.id}`}
+                    data-qa="table-voucher-value-decision-row"
+                  >
+                    <Td>
+                      {decision.child.firstName} {decision.child.lastName}
+                      <br />
+                      <Link to={`/finance/value-decisions/${decision.id}`}>
+                        {`${decision.validFrom.format()} - ${
+                          decision.validTo?.format() ?? ''
+                        }`}
+                      </Link>
+                    </Td>
+                    <Td>{decision.decisionNumber}</Td>
+                    <Td>{formatCents(decision.voucherValue)}</Td>
+                    <Td>{formatCents(decision.finalCoPayment)}</Td>
+                    <Td>{formatDate(decision.created)}</Td>
+                    <Td>{formatDate(decision.sentAt)}</Td>
+                    <Td>{i18n.valueDecision.status[decision.status]}</Td>
+                  </Tr>
+                )
+              })}
+            </Tbody>
+          ))
+          .getOrElse(null)}
+      </Table>
+      {voucherValueDecisions.mapAll({
+        loading() {
+          return <Loader />
+        },
+        failure() {
+          return <div>{i18n.common.loadingFailed}</div>
+        },
+        success() {
+          return null
+        }
+      })}
```

Comment:
huh, finally I understand how this `mapAll` function works, cool. Any reason you why you handle the success case separately instead of moving the rendering code here?

Reply:
Yeah, it's because the elements returned by `loading` and `failure` are not valid children of a `<table>` element. I was thinking of just putting the whole table inside the `success` branch.